### DISCUSSION TITLE
how to show proper exception in VBScriptEngine

### DISCUSSION THREAD

**Discussion by Narasimhara** [ClearFoundry/ClearScript#336] [Q&A] [answered]
hi All,

We have some scripts were there are exceptions. so we have to show proper exceptions. Could you please let me know how to show proper exceptions:

Actual Code:
try
            {
                engine.Invoke(FunctionName);
            }
            catch (Exception e)
            {
                MessageBox.Show(e.Message); // not showing proper message
            } 

the expectation is:

![Untitled2](https://user-images.githubusercontent.com/32035884/148185010-f1e3402a-0c33-496c-8625-1d3b0ba65c7f.png)

**Comment by ClearScriptLib** [comment]
Hi @Narasimhara,

You could try something like this:

``` C#
catch (Exception exception) {
    var baseException = exception.GetBaseException();
    if (baseException is IScriptEngineException scriptEngineException) {
        MessageBox.Show(scriptEngineException.ErrorDetails);
    } else {
        MessageBox.Show(baseException.Message);
    }
}
```

Good luck!

**Reply by Narasimhara** [reply]
Thanks for the help. but it is showing as below, not getting exact information. Could you please help me in this.

![image](https://user-images.githubusercontent.com/32035884/148274675-365e7cbf-c5de-42e9-b1e7-34cab8ca1a0c.png)

**Comment by ClearScriptLib** [comment, marked as answer]
Hmm, it's very odd that there's no error message. Can you provide a minimal sample of the failing code?

As for the stack trace, Windows script engines only provide detailed error information when debugging is enabled. Are you passing `WindowsScriptEngineFlags.EnableDebugging` to the script engine constructor?

Thanks!

**Reply by Narasimhara** [reply]
Thanks for the help after adding  EnableDebugging to true it worked for me. Thanks so much for the help.
